src/main/java/gyro/google/compute/DiskResource.java

```diff
@@ -0,0 +1,288 @@
+            ? resourcePolicies.stream()
+                .map(policy -> ComputeUtils.toResourcePolicyUrl(getProjectId(), policy, getZone().substring(0, getZone().lastIndexOf("-"))))
+                .collect(Collectors.toList())
+            : null);
+    }
+
+    @Override
+    public void setType(String type) {
+        super.setType(type != null ? ComputeUtils.toZoneDiskTypeUrl(getProjectId(), type, getZone()) : null);
+    }
+
+    @Override
+    public void copyMore(Disk disk) {
+        setZone(disk.getZone().substring(disk.getZone().lastIndexOf("/") + 1));
+        setSourceImage(disk.getSourceImage());
+        setSourceImageId(disk.getSourceImageId());
+        setType(disk.getType());
+        setResourcePolicies(disk.getResourcePolicies());
+
+        if (disk.getSourceImageEncryptionKey() != null) {
+            EncryptionKey sourceImageEncryption = newSubresource(EncryptionKey.class);
+            sourceImageEncryption.copyFrom(disk.getSourceImageEncryptionKey());
+            setSourceImageEncryptionKey(sourceImageEncryption);
+        }
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state, Disk disk) {
+        Compute client = createComputeClient();
+
+        disk.setZone(getZone());
+        disk.setSourceImage(getSourceImage());
+        disk.setSourceImageEncryptionKey(getSourceImageEncryptionKey() != null ? getSourceImageEncryptionKey().toCustomerEncryptionKey() : null);
+
+        try {
+            Compute.Disks.Insert insert = client.disks().insert(getProjectId(), getZone(), disk);
+            Operation operation = insert.execute();
+            Operation.Error error = waitForCompletion(client, operation);
+            if (error != null) {
+                throw new GyroException(error.toPrettyString());
+            }
+
+            refresh();
+        } catch (Exception ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public List<ValidationError> validateMore() {
+        return new ArrayList<>();
+    }
+
+    @Override
+    public boolean refresh() {
+        Compute client = createComputeClient();
+
+        try {
+            Disk disk = client.disks().get(getProjectId(), getZone(), getName()).execute();
+            copyFrom(disk);
+
+            return true;
+        } catch (GoogleJsonResponseException je) {
+            if (je.getDetails().getMessage().matches("The resource (.*) was not found")) {
+                return false;
+            } else {
+                throw new GyroException(je.getDetails().getMessage());
+            }
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) {
+        Compute compute = createComputeClient();
+
+        try {
+            compute.disks().delete(getProjectId(), getZone(), getName()).execute();
+        } catch (IOException ex) {
```

Comment:
Should catch the `GoogleJsonResponseException`, which has better details of the error message.
```suggestion
        } catch (GoogleJsonResponseException ex) {
```